src/components/organisms/LocalSearch.tsx

```diff
@@ -0,0 +1,932 @@
+    setNumberOfFoundItems(0)
+  }, [codeMirror])
+
+  const scrollToLine = useCallback((editor, lineNumber: number) => {
+    const t = editor.charCoords({ line: lineNumber, ch: 0 }, 'local').top
+    const middleHeight = editor.getScrollerElement().offsetHeight / 2
+    editor.scrollTo(null, t - middleHeight - 5)
+  }, [])
+
+  const focusSearchItem = useCallback(
+    (
+      editor: CodeMirror.EditorFromTextArea,
+      markers: TextMarker[],
+      selectedIndex: number,
+      focusingEditor = false,
+      updateCursor = false
+    ) => {
+      if (selectedIndex >= markers.length || selectedIndex < 0) {
+        console.warn(
+          'Cannot focus editor on selected item.',
+          selectedIndex,
+          markers.length
+        )
+        return
+      }
+
+      const marker = markers[selectedIndex]
+      if (marker == null) {
+        return
+      }
+      const markerPosition: MarkerRange = marker.find() as MarkerRange
+      if (markerPosition == null) {
+        return
+      }
+      const focusLocation = {
+        line: markerPosition.to.line,
+        ch: markerPosition.to.ch,
+      }
+
+      if (!focusingEditor) {
+        scrollToLine(editor, focusLocation.line)
+      }
+      if (updateCursor) {
+        editor.focus()
+        editor.setCursor(focusLocation.line, focusLocation.ch, {
+          scroll: true,
+        })
+        if (focusingEditor && onCursorActivity != null) {
+          onCursorActivity(editor)
+        }
+      }
+    },
+    [scrollToLine, onCursorActivity]
+  )
+
+  const markFoundItems = useCallback(
+    (codeEditor, selectedItemIdx = -1, regExp) => {
+      const cursor = codeEditor.getSearchCursor(regExp)
+      let first = true
+      let from: CodeMirror.Position
+      let to: CodeMirror.Position
+      let currentItemIndex = 0
+      let previousLine = -1
+      let lineChanged = false
+      while (cursor.findNext()) {
+        from = cursor.from()
+        to = cursor.to()
+
+        if (first) {
+          previousLine = from.line
+          first = false
+        }
+
+        lineChanged = from.line != previousLine
+        previousLine = from.line
+        if (LOCAL_MERGE_SAME_LINE_RESULTS_INTO_ONE) {
+          if (lineChanged) {
+            currentItemIndex++
+          }
+        }
+
+        codeEditor.markText(from, to, {
+          className:
+            currentItemIndex === selectedItemIdx ? 'marked selected' : 'marked',
+          attributes: { dataId: '' + currentItemIndex },
+        })
+        if (!LOCAL_MERGE_SAME_LINE_RESULTS_INTO_ONE) {
+          currentItemIndex++
+        }
+      }
+
+      setNumberOfFoundItems(() => codeEditor.getAllMarks().length)
+    },
+    []
+  )
+
+  const navigateToNextItem = useCallback(
+    (
+      direction: SearchResultNavigationDirection,
+      focusingSearchInput = true
+    ) => {
+      if (loadingResults) {
+        return
+      }
+      let newSelectedItemId: number = selectedItemIndex
```

Comment:
Avoid using `let`. I think we can extract the conditional statement as a function

```ts
const newSelectedItemIndex = getNewSelectedItemIndex(selectedItemIndex, numberOfMarkers, direction)

...

function getNewSelectedItemIndex(currentSelectedIndex: number, numberOfMarkers: number, direction: 'previous' | ...) {...}
```

Reply:
Okay, thanks, refactored, looks way nicer now!